main/inc/lib/hook/HookInterfaces.php

```diff
@@ -0,0 +1,293 @@
+     */
+    public function detachAll();
+
+    /**
+     * Return true if HookManagement plugin is active. Else, false.
+     * This is needed to store attachments into Database inside Hook plugin tables
+     * @return boolean
+     */
+    public static function isHookPluginActive();
+
+    /**
+     * Hook Auto Loader. Search for Hook Observers from plugins
+     * @param string $observerClass
+     * @param string $path
+     * @return int
+     */
+    public static function autoLoadHooks($observerClass, $path);
+}
+
+interface HookObserverInterface
+{
+    /**
+     * Return the singleton instance of Hook observer.
+     * If Hook Management plugin is not enabled, will return NULL
+     * @return HookEventInterface|null
+     */
+    public static function create();
+
+    /**
+     * Return the path from the class, needed to store location or autoload later.
+     * @return string
+     */
+    public function getPath();
+
+    /**
+     * Return the plugin name where is the Hook Observer.
+     * @return string
+     */
+    public function getPluginName();
+}
+
+interface HookManagementInterface
+{
+
+    /**
+     * Initialize Database storing hooks (events, observers, calls)
+     * This should be called right after installDatabase method
+     * @return int
+     */
+    public function initDatabase();
+
+    /**
+     * Insert hook into Database. Return insert id
+     * @param string $eventName
+     * @param string $observerClassName
+     * @param int $type
+     * @return int
+     */
+    public function insertHook($eventName, $observerClassName, $type);
+
+    /**
+     * Delete hook from Database. Return deleted rows number
+     * @param string $eventName
+     * @param string $observerClassName
+     * @param int $type
+     * @return int
+     */
+    public function deleteHook($eventName, $observerClassName, $type);
+
+    /**
+     * Update hook observer order by hook event
+     * @param $eventName
+     * @param $type
+     * @param $newOrder
+     * @return int
+     */
+    public function orderHook($eventName, $type, $newOrder);
+
+    /**
+     * Return a list an associative array where keys are the hook observer class name
+     * @param $eventName
+     * @return array
+     */
+    public function listHookObservers($eventName);
+
+
+    /**
+     * Check if hooks (event, observer, call) exist in Database, if not,
+     * Will insert them into their respective table
+     * @param string $eventName
+     * @param string $observerClassName
+     * @return int
+     */
+    public function insertHookIfNotExist($eventName = null, $observerClassName = null);
+
+
+    /**
+     * Return the hook call id identified by hook event, hook observer and type
+     * @param $eventName
+     * @param $observerClassName
+     * @param $type
+     * @return mixed
+     */
+    public function getHookCallId($eventName, $observerClassName, $type);
+}
+
+/**
+ * Interface HookPluginInterface
+ */
+interface HookPluginInterface
```

Comment:
This interface should be used for plugins to implement hook Observer.
Update documentation